### DISCUSSION TITLE
How to round up formulas in Button outputs

### DISCUSSION THREAD

**Discussion by jdumois01** [mProjectsCode/obsidian-meta-bind-plugin#389] [Q&A] [answered]
tl;dr what syntax do I need to output a rounded number from a Button?

So I'm using Obsidian to store world lore for a TTRPG fantasy setting. There are two dating systems in use, and I use a couple calculations to determine the date of each calendar. When I make a new Historical Event note, I update store the respective dates in the properties as "date1" (Elven Cycle), "date2" (Imperial Calendar), and "date3" (# of Years Ago). Blah blah blah, it's needlessly complicated math. 

I just figured out how to use a Meta Bind Button to do the math and update those properties after I input the "Years Ago" number, but I don't know how to get the calculation to round to the nearest number (or round up to the nearest 1, either is fine). 

![image](https://github.com/user-attachments/assets/a0a0c054-abf9-42cd-a997-0c8fe5b8156c)

I'm brand new to anything resembling coding, so the learning curve is steep. I really appreciate your help! 

Here's my current codeblock:
```meta-bind-button
label: Assimilate Calendar
style: primary
id: calendar
hidden: false
actions:
- type: updateMetadata
  bindTarget: date1
  evaluate: true
  value: 390 - getMetadata('date3') / 12
- type: updateMetadata
  bindTarget: date2
  evaluate: true
  value: 1394 - (getMetadata('date3') * 351.78570709490884/364)
```

**Comment by mProjectsCode** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil

**Reply by jdumois01** [reply]
For some reason I can't use math in the action value formula...where do I need to put this?

**Reply by mProjectsCode** [reply, marked as answer]
probably something like
```yaml
label: Assimilate Calendar
style: primary
id: calendar
hidden: false
actions:
- type: updateMetadata
  bindTarget: date1
  evaluate: true
  value: Math.ceil(390 - getMetadata('date3') / 12)
- type: updateMetadata
  bindTarget: date2
  evaluate: true
  value: Math.ceil(1394 - (getMetadata('date3') * 351.78570709490884/364))
```

**Reply by jdumois01** [reply]
Oh dang, that worked. Apparently I was just missing the capital letter "M". How embarrassing...

Thank you!!
